### ISSUE TITLE
AriocE/baseEncode.cpp 1263: buffer capacity exceeded

### ISSUE THREAD

**Issue by jarrettrios** [RWilton/Arioc#35]
Hello again!

I've been running into the title as an error and can't see why exactly its happening. Other references run fine! 

```
AriocE v1.52.3142.24159 (release) +2024-06-07T20:55:42
Copyright (c) 2015-2024 Johns Hopkins University.  All rights reserved.
 compiled             : 2024-06-13T15:49:58 (GNU g++ v11.4.0)
 data type sizes      : int=4 long=8 *=8 Jvalue5=5 Jvalue8=8 JtableHeader=5 (little-endian)
 computer name        : 94e17f383007
 operating system     : Ubuntu 22.04.4 LTS
 executable file      : /Arioc.x.152/src/AriocE
 configuration file   : /mnt/ramdisk/kraken/experiments/Kraken_Test_V3/encode_gapped_reference.cfg
AriocE::encodeR: encoding 1 file (48 CPU threads available)...
ApplicationException ([0x00000019] AriocE/baseEncode.cpp 1263): buffer capacity exceeded (136/112 bytes)
Command exited with return code 1
```
Here is the cfg and the reference.fa file. I had to add a .txt to the end of the file name to be able to upload them to github

[encode_gapped_reference.cfg.txt](https://github.com/user-attachments/files/15873837/encode_gapped_reference.cfg.txt)
[reference_templates.fa.txt](https://github.com/user-attachments/files/15873838/reference_templates.fa.txt)

**Comment by RWilton** [comment]
Yikes!  AriocE isn't properly digesting that simple FASTA file.

I will dig into this, try to produce a more informative failure message, and fix any bug that I find.

[Update:  I am able to reproduce the error message, but I have not yet identified the problem.  I will post a solution as soon as possible.]

[Update:  The error message is accurate.  It's one of those "this should never happen" sanity checks that does indeed trap an internal buffer allocation error.  I will fix this bug and post an updated "pre-release" build tomorrow.]

**Comment by RWilton** [comment]
This bug is fixed in the current Arioc release (v1.52.3143).

**Comment by jarrettrios** [comment]
Exciting, thank you!
